### PR TITLE
Fully quantized model, precision scans

### PR DESCRIPTION
**Pull request by therwig**
Hi all,
Ben and I have gotten the full qCNN model quantized and added functionality to do scans versus precision. Should be easily adapted / extended if others want to play with precision scans as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*csv
+*png
+test*
+*~
+*pycache*
+data
+log*
```
Comment (kakwok) [inline .gitignore:7]: Thanks for adding this!

File: train.py [hunk 2]
```diff
@@ -30,79 +33,80 @@ def normalize(data,rescaleInputToMax=False):
...
+        encoded_shape = encoded_Q.shape
+        encoded_Q = np.reshape(encoded_Q,(len(encoded_Q),encoded_shape[3],encoded_shape[1]))
+
+    return decoded_Q, encoded_Q
 
 ### cross correlation of input/output 
 def cross_corr(x,y):
     cov = np.cov(x.flatten(),y.flatten())
-    std = np.sqrt(np.diag(cov))
+    std = np.sqrt(np.diag(cov)+1e-10)
```
Comment (kakwok) [inline train.py:109]: this should be without the 1e-10?

**Review by kakwok — CHANGES_REQUESTED** [review]
Thanks for refactoring the code! Just one question of the correlation calcuation

**Comment by therwig** [comment]
Made some more changes in the spirit of #3. Single training script now with `--quantize` flag. I'm still debating whether the best way to pass precision to the training function. Right now theres an optional argument of 'model parameters to update' passed directly to the function that the training script makes use of, but I may add a command line options to control instead.

The scan script is broken out into a different file and more plots are added.

File: train.py (hunk 2 above)
Comment (therwig) [reply train.py:109]: do we need to protect against zero division? that was the idea here

Comment (therwig) [reply train.py:109]: though we could do it in a smarter way

Comment (kakwok) [reply train.py:109]: I see, i guess it would protect cases like 1 epoch test run for completing. We could leave that for now

**Review by kakwok — APPROVED** [review]

**Pull request merged**